### PR TITLE
Small refactoring in rep.c

### PR DESCRIPTION
**Pull request by adizaimi**
Small refactoring in rep.c

### CODE REVIEW

File: bdb/rep.c [hunk 1]
```diff
@@ -2116,11 +2113,12 @@ void bdb_get_notcoherent_list(bdb_state_type *bdb_state,
...
-                    nodes_list[*num_notcoherent] = hostlist[i];
-                (*num_notcoherent)++;
-            }
+            if ((*num_notcoherent) < max_nodes)
+                nodes_list[*num_notcoherent] = hostlist[i];
+            (*num_notcoherent)++;
+
+            if (bdb_state->repinfo->skipsinceepoch == 0)
+                abort();
         }
```
Comment (mistachkin) [inline bdb/rep.c]: There are no circumstances where skipsinceepoch can legitimately be zero here?

File: bdb/rep.c [hunk 2]
```diff
@@ -2843,25 +2848,25 @@ static void bdb_slow_replicant_check(bdb_state_type *bdb_state,
...
-                malloc(sizeof(wait_for_lsn_list));
-            listc_init(bdb_state->seqnum_info->waitlist[nodeix(host)],
-                       offsetof(struct waiting_for_lsn, lnk));
-        }
+    struct waiting_for_lsn *waitforlsn;
+    int node_ix = nodeix(host);
+    if (bdb_state->seqnum_info->waitlist[node_ix] == NULL) {
+        bdb_state->seqnum_info->waitlist[node_ix] =
+            malloc(sizeof(wait_for_lsn_list));
+        listc_init(bdb_state->seqnum_info->waitlist[node_ix],
```
Comment (mistachkin) [inline bdb/rep.c]: Perhaps this malloc should be a calloc?

File: bdb/rep.c (hunk 1 above)
Comment (adizaimi) [reply bdb/rep.c]: I guess i was trying to assert that skipsinceepoch is non zero. This check is not needed, i'm removing it -- we don't gain anything from it. skipsinceepoch is set every time a node is not coherent.

File: bdb/rep.c (hunk 2 above)
Comment (adizaimi) [reply bdb/rep.c]: Subsequent listc_init() sets all members of the listc_t so we don't need calloc here.

**Pull request merged**